libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -516,17 +516,36 @@ public boolean isVcSuccessful(List<VcStatusDto> currentVcStatuses, String criIss
                 .getIsSuccessfulVc();
     }
 
-    private boolean isEvidenceVc(VcStoreItem item) throws CredentialParseException {
-        JsonNode vcEvidenceNode = getVCClaimNode(item.getCredential(), VC_EVIDENCE);
+    private boolean isEvidenceVc(String credential, boolean isNonZero)
+            throws CredentialParseException {
+        JsonNode vcEvidenceNode = getVCClaimNode(credential, VC_EVIDENCE);
         for (JsonNode evidence : vcEvidenceNode) {
             if (evidence.path(VC_EVIDENCE_VALIDITY).isInt()
                     && evidence.path(VC_EVIDENCE_STRENGTH).isInt()) {
+                if (isNonZero
+                        && (evidence.path(VC_EVIDENCE_VALIDITY).asInt() == 0
+                                || evidence.path(VC_EVIDENCE_STRENGTH).asInt() == 0)) {
+                    return false;
+                }
                 return true;
             }
         }
         return false;
     }
 
+    public List<SignedJWT> filterValidCredentials(List<SignedJWT> credentials) {
+        return credentials.stream()
+                .filter(
+                        item -> {
+                            try {
+                                return isEvidenceVc(item.serialize(), true);
```

Comment:
Hmm - it's a bit annoying that we have to keep serializing and deserializing the VC, but hopefully that'll be solved by https://govukverify.atlassian.net/browse/PYIC-3908

Feels like it'll be painful to address now